src/main/java/com/mraof/minestuck/client/gui/ReadableSburbCodeScreen.java

```diff
@@ -0,0 +1,298 @@
+		{
+			int fromSectionIndex = sectionIterate * linesPerBlock;
+			int toSectionIndex = Math.min((sectionIterate * linesPerBlock + linesPerBlock - 1), textList.size());
+			
+			if(!hieroglyphValidityArray[sectionIterate]) //if there was no valid hieroglyph for that element
+			{
+				if(sectionIterate == hieroglyphValidityArray.length - 1)
+					toSectionIndex = textList.size() - 1; //add any overflow to the last section
+				
+				for(int iterate = fromSectionIndex; iterate < toSectionIndex + 1; iterate++)
+				{
+					textList.set(iterate, EMPTY_SPACE);
+				}
+			}
+		}
+	}
+	
+	/**
+	 * Groups the textList lines into pages after textList has been processed by makeUnrecordedSectionsBlank()
+	 */
+	private void sortIntoPages()
+	{
+		for(int pageIterate = 0; pageIterate < totalPages + 1; pageIterate++)
+		{
+			int fromIndex = Math.min(LINES_PER_PAGE * pageIterate, textList.size());
+			int toIndex = Math.min(LINES_PER_PAGE * pageIterate + LINES_PER_PAGE, textList.size());
+			
+			listOfPages.add(textList.subList(fromIndex, toIndex));
+		}
+	}
+	
+	@Override
+	public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTicks)
+	{
+		this.renderBackground(poseStack);
+		this.setFocused(null);
+		RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
+		int pageModulo = currentPage % 3;
+		if(currentPage == 0) //specifically if its the first page
+			RenderSystem.setShaderTexture(0, BOOK_TEXTURES_01A);
+		else if(pageModulo == 0)
+			RenderSystem.setShaderTexture(0, BOOK_TEXTURES_01B);
+		else if(pageModulo == 1)
+			RenderSystem.setShaderTexture(0, BOOK_TEXTURES_02);
+		else if(pageModulo == 2)
+			RenderSystem.setShaderTexture(0, BOOK_TEXTURES_03);
+		
+		
+		int topX = (this.width - GUI_WIDTH) / 2;
+		int topY = 2;
+		this.blit(poseStack, topX, topY, 0, 0, GUI_WIDTH, GUI_HEIGHT);
+		{
+			if(textList != null)
+			{
+				//pushPose function, followed by the scale function and ending with the popPose function, allow the scale changes to only be applied to the text
+				poseStack.pushPose();
+				float subtractScale = 0.4F;
+				float scale = (1 / subtractScale);
+				poseStack.scale(subtractScale, subtractScale, subtractScale);
+				
+				MutableInt lineY = new MutableInt();
+				
+				//takes the necessary page from the group of pages and then reads out each of the 40 lines stored for that page
+				for(String text : listOfPages.get(currentPage))
+				{
+					font.getSplitter().splitLines(text, TEXT_WIDTH, Style.EMPTY, true, (style, start, end) -> {
+						//limiting the length of the page via this if statement
+						if(stillValidLine(lineY.intValue()))
+						{
+							Component line = new TextComponent(text.substring(start, end)).setStyle(style);
+							font.draw(poseStack, line, ((this.width - GUI_WIDTH) / 2F + TEXT_OFFSET_X) * scale, (lineY.intValue() + TEXT_OFFSET_Y) * scale, 0x00A300); //hex is green
+							lineY.add(CUSTOM_LINE_HEIGHT);
+						}
+					});
+				}
+			}
+		}
+		
+		poseStack.popPose();
```

Comment:
Move this inside the same block as the push. A pop must only ever be called when it is certain that its corresponding push has been called.